inginious/frontend/plugins/task_hints/pages/user_hint_manager.py

```diff
@@ -163,6 +188,119 @@ def update_total_penalty(self, task_id, username):
                                                       {"$set": {"total_penalty": new_penalty}
                                                        })
 
+    def on_change_task_submission_mode(self, courseid, task_id, submission_mode_change, task_hints):
+
+        "Update the user hints document structure when the task mode is changed."
+        "When a individually task, each student have it's own user hints document."
+        "When a group task, there is only a document for the students group"
+
+        if(submission_mode_change):  
+            
+            classrooms = self._database.aggregations.find({"courseid": courseid}
+            )
+
+            for classroom in classrooms:
```

Comment:
This `for` is not necessary, is it? The query will always return empty or the course, this collection cannot have two documents with the same courseid